server/s3_object_metadata.cc

```diff
@@ -35,126 +37,124 @@
-  layout_id = 0;
-  old_layout_id = 0;
-  tried_count = 0;
 
   object_key_uri = bucket_name + "\\" + object_name;
 
   // Set the defaults
-  system_defined_attribute["Date"] = "";
-  system_defined_attribute["Content-Length"] = "";
-  system_defined_attribute["Last-Modified"] = "";
-  system_defined_attribute["Content-MD5"] = "";
+  (void)system_defined_attribute["Date"];
+  (void)system_defined_attribute["Content-Length"];
+  (void)system_defined_attribute["Last-Modified"];
+  (void)system_defined_attribute["Content-MD5"];
+
   // Initially set to requestor of API, but on load json can be overriden
   system_defined_attribute["Owner-User"] = user_name;
   system_defined_attribute["Owner-Canonical-id"] = canonical_id;
   system_defined_attribute["Owner-User-id"] = user_id;
   system_defined_attribute["Owner-Account"] = account_name;
   system_defined_attribute["Owner-Account-id"] = account_id;
 
-  system_defined_attribute["x-amz-server-side-encryption"] =
-      "None";  // Valid values aws:kms, AES256
-  system_defined_attribute["x-amz-version-id"] =
-      "";  // Generate if versioning enabled
-  system_defined_attribute["x-amz-storage-class"] =
-      "STANDARD";  // Valid Values: STANDARD | STANDARD_IA | REDUCED_REDUNDANCY
-  system_defined_attribute["x-amz-website-redirect-location"] = "None";
-  system_defined_attribute["x-amz-server-side-encryption-aws-kms-key-id"] = "";
-  system_defined_attribute["x-amz-server-side-encryption-customer-algorithm"] =
-      "";
-  system_defined_attribute["x-amz-server-side-encryption-customer-key"] = "";
-  system_defined_attribute["x-amz-server-side-encryption-customer-key-MD5"] =
-      "";
+  str_set_default(system_defined_attribute["x-amz-server-side-encryption"],
+                  "None");
+  (void)system_defined_attribute["x-amz-version-id"];
+
+  // Generate if versioning enabled
+  str_set_default(system_defined_attribute["x-amz-storage-class"], "STANDARD");
+  str_set_default(system_defined_attribute["x-amz-website-redirect-location"],
+                  "None");
+
+  (void)system_defined_attribute["x-amz-server-side-encryption-aws-kms-key-id"];
+  (void)system_defined_attribute
+      ["x-amz-server-side-encryption-customer-algorithm"];
+  (void)system_defined_attribute["x-amz-server-side-encryption-customer-key"];
+  (void)
+      system_defined_attribute["x-amz-server-side-encryption-customer-key-MD5"];
+
   if (is_multipart) {
     index_name = get_multipart_index_name();
     system_defined_attribute["Upload-ID"] = upload_id;
-    system_defined_attribute["Part-One-Size"] = "";
+    (void)system_defined_attribute["Part-One-Size"];
   } else {
     index_name = get_object_list_index_name();
   }
-
-  object_list_index_oid = {0ULL, 0ULL};
-  objects_version_list_index_oid = {0ULL, 0ULL};
 }
 
 S3ObjectMetadata::S3ObjectMetadata(
-    std::shared_ptr<S3RequestObject> req, bool ismultipart,
-    std::string uploadid,
+    std::shared_ptr<S3RequestObject> req, std::string bucketname,
```

Comment:
pass the bucket name and object name as const references.

Reply:
I thought about it. Passing of objects is less expensive in our case.

Reply:
I got your point, but still its good practice to pass strings as const references, unless we want to update them in the function. 

We are open source, so following standard coding practices will give our code good reputation among open-source community.

Reply:
Here the strings are passed by value, not by non-const reference.
Amit, trust me here, I exactly know what I've done.